src/Illuminate/Broadcasting/Broadcasters/Broadcaster.php

```diff
@@ -73,14 +73,63 @@ protected function verifyUserCanAccessChannel($request, $channel)
 
             $handler = $this->normalizeChannelHandlerToCallable($callback);
 
-            if ($result = $handler($this->retrieveUser($request, $channel), ...$parameters)) {
+            $user = $this->retrieveUser($request, $channel);
+
+            if (! $this->canBeCalledWithUser($user, $handler)) {
+                continue;
+            }
+
+            if ($result = $handler($user, ...$parameters)) {
                 return $this->validAuthenticationResponse($request, $result);
             }
         }
 
         throw new AccessDeniedHttpException;
     }
 
+    /**
+     * Determine whether the callback can be called with the given user.
+     *
+     * @param  \Illuminate\Contracts\Auth\Authenticatable|null  $user
+     * @param  \Closure|callable $callback
```

Comment:
`\Closure` is already callable, so `@param  callable  $callback` is enough

Reply:
Thanks!